### PR TITLE
Upstream changes for v0.5.0 release

### PR DESCRIPTION
**Pull request by shubhadeepd**
These changes adds new dedicated RAG examples showcasing state of the art usecases, switches to the latest [API catalog endpoints from NVIDIA](https://build.nvidia.com/explore/discover) and also refactors the API interface of chain-server. This release also improves the developer experience by adding github pages based documentation and streamlining the example deployment flow using dedicated compose files.

Check the `changelog.md` file for the detailed changes.

### CODE REVIEW

File: RetrievalAugmentedGeneration/examples/multi_turn_rag/requirements.txt
Comment (sumitkbh) [inline RetrievalAugmentedGeneration/examples/multi_turn_rag/requirements.txt:1]: Why a empty file is pushed?

File: experimental/product-advisor/.dockerignore
Comment (sumitkbh) [inline experimental/product-advisor/.dockerignore:1]: product-advisor RAG cannot be published. Remove it from this PR

**Review by mikemckiernan — APPROVED** [review]
Other than the fact that it's a doozie of a PR and Sumit's comment, LGTM.  TY.

File: RetrievalAugmentedGeneration/examples/multi_turn_rag/requirements.txt
Comment (shubhadeepd) [reply RetrievalAugmentedGeneration/examples/multi_turn_rag/requirements.txt:1]: Removed.

**Pull request merged**